app/models/game.rb

```diff
@@ -29,29 +29,210 @@ def populate_board!
+  # Query for remaining opponent pieces
+    @remaining_opponent_pieces = pieces.where(game_id: id, captured?: false).where.not(color: color)
+  # Grab the king from @remaining_opponent_pieces
+    @opponent_king = @remaining_opponent_pieces.select { |remaining_opponent_piece| remaining_opponent_piece.type.eql? 'King' }.last
+
+  # Store opponent_king coordinates for shorter reference
+    @okx = @opponent_king.x_coordinate
+    @oky = @opponent_king.y_coordinate
+  end
+  
+  def set_coordinates_between_piece_causing_check_and_opponent_king
+  # Store coordinates of player piece causing check locally
+    x_of_threat = @x_of_threat
+    y_of_threat = @y_of_threat
+  
+  # Initialize array that will hold coordinate sets between player piece causing check and opponent king
+    coordinate_path_to_king = []
+    
+  # If the x coordinate of the piece causing check matches the opponent king x
+  # coordinate, build a coordinate path in which only the y coordinate needs to change
+    if @x_of_threat == @okx
+      if @y_of_threat > @oky
+        @y_of_threat.downto(@oky).each { |y| coordinate_path_to_king << [x_of_threat, y] }
+      else
+        @y_of_threat.upto(@oky).each { |y| coordinate_path_to_king << [x_of_threat, y] }
+      end
+  # If the y coordinate of the piece causing check matches the opponent king y
+  # coordinate, build a coordinate path in whch only the x coordinate needs to change
+    elsif @y_of_threat == @oky
+      if @x_of_threat > @okx
+        @x_of_threat.downto(@okx).each { |x| coordinate_path_to_king << [x, y_of_threat] }
+      else
+        @x_of_threat.upto(@okx).each { |x| coordinate_path_to_king << [x, y_of_threat] }
+      end
+  # If neither the x or y coordinates of the piece causing check match the opponent king x or y
+  # coordinates, build a diagonal coordinate path
+    else
+      if @x_of_threat < @okx
+        if @y_of_threat > @oky
+          @x_of_threat.upto(@okx).each do |x|
+            coordinate_path_to_king << [x, y_of_threat]
+            y_of_threat -= 1
+          end
+        else
+          @x_of_threat.upto(@okx).each do |x|
+            coordinate_path_to_king << [x, y_of_threat]
+            y_of_threat += 1
+          end
+        end
+      else
+        if @y_of_threat > @oky
+          @x_of_threat.downto(@okx).each do |x|
+            coordinate_path_to_king << [x, y_of_threat]
+            y_of_threat -= 1
+          end
+        else
+          @x_of_threat.downto(@okx).each do |x|
+            coordinate_path_to_king << [x, y_of_threat]
+            y_of_threat += 1
+          end
+        end
+      end
+    end
+    
+  # Return coordinate_path_to_king without the first and last sets of coordinates
+  # (i.e. where the player piece is moving & where the opponent king is)
+    coordinate_path_to_king.shift
+    coordinate_path_to_king.pop
+    coordinate_path_to_king
+  end
+  
+  def is_finished?
```

Comment:
Heh, this appears to be similar to our recent discussion about 'is_obstructed?'. I certainly appreciate that there's a 'style guide,' but that word 'guide' is rather key. A guide is just that, not a series of hard and fast rules. I simply find that what I have there is more readable, and I don't feel compelled to drop three characters just because it's suggested to be redundant. A matter of opinion, no?